src/Raven.Server/Documents/Replication/ResolveConflictOnReplicationConfigurationChange.cs

```diff
@@ -400,6 +406,57 @@ public DocumentConflict ResolveToLatest(List<DocumentConflict> conflicts)
 
             return latestDoc;
         }
+
+        private void ResolveAttachmentsConflicts(DocumentsOperationContext context, DocumentConflict resolved)
+        {
+            using var scope = Slice.External(context.Allocator, resolved.LowerId, out var lowerId);
+            var storageAttachmentsDetails = _database.DocumentsStorage.AttachmentsStorage.GetAttachmentDetailsForDocument(context, lowerId);
+            List<AttachmentName> resolvedAttachmentsMetadata = null;
+            bool resolveToLatest = ScriptConflictResolversCache.TryGetValue(resolved.Collection, out ScriptResolver scriptResolver) == false || scriptResolver == null;
+
+            foreach (var group in storageAttachmentsDetails.GroupBy(x => x.Name))
+            {
+                if (group.Count() == 1)
+                    continue;
+
+                resolvedAttachmentsMetadata ??= GetAttachmentsFromMetadata(resolved.Doc);
+                var found = false;
+
+                foreach (var attachment in group)
+                {
+                    if (found == false && resolvedAttachmentsMetadata.Any(x => x.Name == attachment.Name && x.Hash == attachment.Hash && x.ContentType == attachment.ContentType))
```

Comment:
I don't understand what you are doing here, can you explain?

Reply:
In the storage we already have all attachments for particular document id (both conflict + local).
In resolved document we have the attachments metadata which can be modified in script resolver.

I get the attachments from storage to `storageAttachmentsDetails` then group by name, if I have more then 1 attachment in group, this mean I have duplicate attachments. Then I retrieve the `@metadata.@attachments` for resolved document and for each attachment in the group (duplicates) I check if I have the attachment in resolved metadata, if I do I mark it as `found` (so I don't have to check the resolved attachment so all next duplicates), for each next duplicate I "resolve" it according to the next logic: 
1. if `conflictResovler` doesn't have script I delete all duplicates, this mean only the attachment found in resolved metadata will be kept.
2. if  `conflictResovler` do have a script I keep all the duplicates and rename them, by this we don't let the use modify the attachment using the resolver script.

Reply:
My issue is that you aren't syncing between the storage and document attachments.
And why are you comparing to the document values on hash & content type? What does the script give you in this case?

Reply:
The storage already have all attachments from all conflicts, when we put the resolved doc we recreate the attachments metadata from storage
https://github.com/ravendb/ravendb/blob/v4.2/src/Raven.Server/Documents/DocumentPutAction.cs#L395

In this method I retrieve the attachments from storage so I have the merged attachments list, and manipulate the according to conflict resolution strategy.

I compare the attachment from storage to attachment from resolved doc's metadata by name, hash, key (those 3 are part of the attachment key in the table). In this way I can know that I don't have to delete/rename this attachment in storage because it is in resolved metadata.